app/models/ems_refresh/parsers/kubernetes.rb

```diff
@@ -121,9 +121,13 @@ def parse_node(node)
         parse_node_condition(condition)
       end
 
-      # Searching for the underlying instance. At the moment this search strategy
-      # is tested only with OpenStack.
-      vms = VmOpenstack.where(:uid_ems => new_result[:identity_system].downcase)
+      # Establish a relationship between this node and the vm its on (if its in the system)
+      # supported relationships: oVirt and Openstack
+      types = [ManageIQ::Providers::Redhat::InfraManager::Vm.name,
+               VmOpenstack.name]
+
+      # Searching for the underlying instance for Openstack or oVirt.
+      vms = Vm.where(:uid_ems => new_result[:identity_system].downcase, :type => types)
```

Comment:
```sql
explain SELECT "vms".* FROM "vms" 
WHERE "vms"."type" IN ('Vm', 'ManageIQ::Providers::InfraManager::Vm', 'ManageIQ::Providers::CloudManager::Vm', 'VmServer', 'ManageIQ::Providers::Vmware::InfraManager::Vm', 'ManageIQ::Providers::Redhat::InfraManager::Vm', 'VmMicrosoft', 'VmXen', 'ManageIQ::Providers::Amazon::CloudManager::Vm', 'VmOpenstack') 
AND  "vms"."type"  IN ('ManageIQ::Providers::Redhat::InfraManager::Vm', 'VmOpenstack') ;
```
generates
```sql
Filter: (((type)::text = ANY ('{ManageIQ::Providers::Redhat::InfraManager::Vm,VmOpenstack}'::text[])) 
     AND ((type)::text = ANY ('{Vm,ManageIQ::Providers::InfraManager::Vm,ManageIQ::Providers::CloudManager::Vm,VmServer,ManageIQ::Providers::Vmware::InfraManager::Vm,ManageIQ::Providers::Redha
t::InfraManager::Vm,VmMicrosoft,VmXen,ManageIQ::Providers::Amazon::CloudManager::Vm,VmOpenstack}'::text[])))
```
My guess is that there  probably is some additional overhead head here.